util/design/secded_gen.py

```diff
@@ -243,43 +238,190 @@ def hamming_code(n, k, m):
                 # If even, we are in the skip phase, do not include
                 # If odd, we are in the include phase
                 parity_chk = int((pos - (pos % parity_pos)) / parity_pos)
-                log.debug("At position {} parity value {}, {}" \
-                         .format(pos, parity_pos, parity_chk))
+                log.debug("At position {} parity value {}, {}".format(
+                    pos, parity_pos, parity_chk))
 
                 # valid for inclusion or final parity bit that includes everything
-                if is_odd(parity_chk) or p == m-1:
-                    code = code + (p,)
+                if is_odd(parity_chk) or p == m - 1:
+                    code = code + (p, )
                     log.info("add {} to tuple {}".format(p, code))
 
             codes.append(code)
 
+    # final parity bit includes all ECC bits
+    for p in range(m - 1):
```

Comment:
ah thanks for catching this, i completely didn't notice i forgot to include the other parity bits.

Reply:
no worries. 
the bug is not visible until you actually introduce an error, and we haven't run these tests so far. 
also, the syndrome decoding and data correction is the same for Hamming and Hsiao, so the bug does not show up in the data bits.